elements/howto-accordion/howto-accordion.js

```diff
@@ -0,0 +1,533 @@
+     */
+    _allPanels() {
+      return Array.from(this.querySelectorAll('howto-accordion-heading + *'));
+    }
+
+    /**
+     * `_allHeadings` returns all the headings in the accordion.
+     */
+    _allHeadings() {
+      return Array.from(this.querySelectorAll('howto-accordion-heading'));
+    }
+
+    /**
+     * `_panelForHeading` returns the panel that the given heading controls.
+     */
+    _panelForHeading(heading) {
+      return heading.nextElementSibling;
+    }
+
+    /**
+     * `_prevHeading` returns the heading that comes before the currently
+     * selected one, wrapping around when reaching the first one.
+     */
+    _prevHeading() {
+      const headings = this._allHeadings();
+      // Use `findIndex` to find the index of the currently
+      // selected element and subtracts one to get the index of the previous
+      // element.
+      let newIdx =
+        headings.findIndex(headings =>
+          headings === document.activeElement.parentElement) - 1;
+      // Add `headings.length` to make sure the index is a positive number
+      // and get the modulus to wrap around if necessary.
+      return headings[(newIdx + headings.length) % headings.length];
+    }
+
+    /**
+     * `_nextHeading` gets the heading that comes after the currently selected
+     * one, wrapping around when reaching the last heading.
+     */
+    _nextHeading() {
+      const headings = this._allHeadings();
+      let newIdx =
+        headings.findIndex(heading =>
+          heading === document.activeElement.parentElement) + 1;
+      return headings[newIdx % headings.length];
+    }
+
+    /**
+     * `_firstHeading` returns the first heading.
+     */
+    _firstHeading() {
+      const headings = this._allHeadings();
+      return headings[0];
+    }
+
+    /**
+     * `_lastHeading` returns the last heading.
+     */
+    _lastHeading() {
+      const headings = this._allHeadings();
+      return headings[headings.length - 1];
+    }
+
+    /**
+     * `_expandPanel` puts the given panel in the expanded state, without any
+     * animation.
+     */
+    _expandPanel(panel) {
+      panel.setAttribute('aria-hidden', 'false');
+      panel.classList.remove('collapsed');
+      panel.classList.add('expanded');
+    }
+
+    /**
+     * `_expandHeading` puts the given heading in the expanded state, without
+     * any animation.
+     */
+    _expandHeading(heading) {
+      heading.setAttribute('aria-expanded', 'true');
+      heading.classList.add('expanded');
+    }
+
+    /**
+     * `_expand` expands both the given heading and panel using an animation.
+     */
+    _expand(heading, panel) {
+      // Put both the heading and the panel in the expanded state immediately.
+      // The animation function will hide the panel.
+      this._expandHeading(heading);
+      this._expandPanel(panel);
+
+      this._animateIn(panel);
+    }
+
+    _collapsePanel(panel) {
+      panel.setAttribute('aria-hidden', 'true');
```

Comment:
Seeing stuff like this makes me wonder if it's not better to just create a `<howto-accordion-panel>` element that could do all this stuff in response to a property or attribute being set. Right now similar logic lives inside the `<howto-accordion-heading>` so this element has a sort of split personality when it comes to its responsibilities.